### ISSUE TITLE
OmniStudio Compatibility

### ISSUE THREAD

**Issue by bensezio** [Codeneos/vlocode#392]
Is it possible to make this work for the Omnistudio namespace? What are the possible configuration options to make this happen?

Many thanks.

**Comment by Codeneos** [comment]
I think it should be. I don't have a test environment with the Omnistudio package to test yet but the extension was written keeping support for Vlocity packages with different namespaces in mind.

Right now I think there is one check that queries the installed packages and looks for a package with Vlocity in the name that might be blocking this.

What is the developer name of the Omnistudio package?

**Comment by wwwmonkey** [comment]
G'day @Codeneos you can now (at your leisure) signup for a Developer edition with Omnistudio here:
https://trailhead.salesforce.com/promo/orgs/omnistudiotrails

By developer name do you mean Publisher? It's "Salesforce" in my Org.

This is the error received when attempting to deploy a DR today:
` [VlocodeService] [error] Error: No such object with name %vlocity_namespace%__DRMatchingKey__mdt exists in this Salesforce instance`

Please lmk if you need any assistance troubleshooting etc as I really appreciate your work on this plugin.

**Comment by Codeneos** [comment]
@wwwmonkey  thanks for info! I'll have a look into this. The new setting that you pointed out will be useful as a fallback to detect to get the namespace.

Once I have some more details I will get back and try to update the plugin so it is also compatible with OmniStudio.

**Comment by iamlesterjoseph** [comment]
Hello @Codeneos Has this been resolved? I was trying to use the Vlocity extension on VSCode but I am having the same error. I cannot connect to the org that I am using. And the org has "omnistudio" package namespace installed.

**Comment by Codeneos** [comment]
No not yet, but I am working on updating Vlocode to support OmniStudio. 

I am aiming to have a beta version out by end of this week depending on how much time I get to work on this :)

**Comment by mskksfdc** [comment]
this is the YAML we had to build to get vlocity components within omnistudio namespace

projectPath: ./vlocity-tmp
queries: # Queries to get vlocity datapacks
  - VlocityDataPackType: OmniScript
    query: Select Id,Type,SubType,Language,OmniProcessType,Description,VersionNumber,IsActive from OmniProcess Where OmniProcessType ='OmniScript' and IsActive = true and Type='' AND SubType =''
manifestOnly: true
activate: true
maxDepth: -1
defaultMaxParallel: 100
gitCheck: true # Git Hash of previous deployment stored in Setup-->Custom Settings-->General Settings
delete: true # Delete the VlocityDataPack__c file on finish
autoUpdateSettings: true

**Comment by hassanahmed0071** [comment]
@Codeneos  is this fixed ? Does Vlocode now supports Omnistudio ?  Because I am getting same errors.

**Comment by Codeneos** [comment]
@hassanahmed0071 no, not yet - as I'm personally not yet using OmniStudio I haven't been able to put in the work to make it compatible. There is a test branch but that only has limited support at the moment.

I do intend to add full support for OmniStudio and have and now also have a test org OmniStudio enabled.

**Comment by Codeneos** [comment]
Better late then never, OmniStudio is now fully supported in version 2.0.0 of the extension.
